src/gei/Commands/MigrateArchiveRepoCommand.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class MigrateArchiveRepoCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+        private readonly IAzureApiFactory _azureApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        private const int ARCHIVE_GENERATION_TIMEOUT_IN_HOURS = 10;
+        private const int CHECK_STATUS_DELAY_IN_MILLISECONDS = 10000; // 10 seconds
+        private const string GIT_ARCHIVE_FILE_NAME = "gitArchive.tar.gz";
+        private const string METADATA_ARCHIVE_FILE_NAME = "metadataArchive.tar.gz";
+
+        public MigrateArchiveRepoCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, ITargetGithubApiFactory targetGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider, IAzureApiFactory azureApiFactory) : base("migrate-archive-repo")
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _azureApiFactory = azureApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+            _targetGithubApiFactory = targetGithubApiFactory;
+
+            Description = "Generates migration archives, uploads them to Azure Blob Storage, then invokes the GitHub Migration APIs to migrate the repo and all repo data using those uploaded archives.";
```

Comment:
I think we need a bit more info in here. Customers are going to see this command side-by-side with migrate-repo and not know which one to pick.

Reply:
Will be removed in https://github.com/github/gh-gei/pull/221#discussion_r801977286

Reply:
Sure but in the interim customers are going to see this in the next monday release and potentially be confused. Couldn't we add a simple description, even if it's telling customers it's a work-in-progress and they should not use it yet.

Reply:
Updated in https://github.com/github/gh-gei/pull/221/commits/5d823ca5d3072fb86da374fad2b6d2646c69ce4f